osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -353,6 +356,52 @@ protected override void HandleItemSelected(object? model)
             }
         }
 
+        protected override bool HandleItemsChanged(NotifyCollectionChangedEventArgs args)
+        {
+            switch (args.Action)
+            {
+                case NotifyCollectionChangedAction.Add:
+                case NotifyCollectionChangedAction.Remove:
+                case NotifyCollectionChangedAction.Move:
+                case NotifyCollectionChangedAction.Reset:
+                    return true;
+
+                case NotifyCollectionChangedAction.Replace:
+                    var oldBeatmaps = args.OldItems!.OfType<BeatmapInfo>().ToList();
+                    var newBeatmaps = args.NewItems!.OfType<BeatmapInfo>().ToList();
+
+                    for (int i = 0; i < oldBeatmaps.Count; i++)
+                    {
+                        var oldBeatmap = oldBeatmaps[i];
+                        var newBeatmap = newBeatmaps[i];
+
+                        // Ignore changes which don't concern us.
+                        //
+                        // Here are some examples of things that can go wrong:
+                        // - Background difficulty calculation runs and causes a realm update.
+                        //   We use `BeatmapDifficultyCache` and don't want to know about these.
+                        // - Background user tag population runs and causes a realm update.
+                        //   We don't display user tags so want to ignore this.
+                        if (
+                            // covers metadata changes
+                            oldBeatmap.Hash == newBeatmap.Hash &&
+                            // displayed
+                            oldBeatmap.Status == newBeatmap.Status &&
+                            // displayed
+                            oldBeatmap.DifficultyName == newBeatmap.DifficultyName &&
+                            // sanity
+                            oldBeatmap.OnlineID == newBeatmap.OnlineID
+                        )
```

Comment:
The test failures show that this cannot really be applied to filtering like this unless you also include *all possible fields the user could conceivably filter song select by*. And that, among others, includes user tags, date ranked, date submitted, all of which are fields that could be continually getting molested by background backpopulations. Which is not something that I realised, admittedly.

Reply:
It's probably the best that can be done, but it'll also be probably very ugly...

Reply:
Actually for something like user tags, i would be fine with it not re-filtering on background update.

The failing tests are for hiding, which should be simple enough to fix (i hope (but probably isn't (and you probably already know this))).

Reply:
> LastPlayed is another one which is likely to cause similar trouble here.

Possibly an exception which can be addressed by filtering on returning from gameplay, dunno.

Reply:
Yep, I'll look into this, and *maybe* add test coverage.